src/client/java/daos-java/src/main/native/io_daos_obj_DaosObjClient.c

```diff
@@ -249,8 +249,8 @@ decode_initial(data_desc_t *desc, char *desc_buffer)
 	uint64_t value64;
 	int i;
 
-	if (desc->iods == NULL || desc->sgls == NULL
-		|| desc->recxs == NULL || desc->iovs == NULL) {
+	if (desc->iods == NULL | desc->sgls == NULL
```

Comment:
Just wondering why you changed "||" to "|" ? For C code styles, we normally did it using "||" to do or check condition.

Reply:
> Just wondering why you changed "||" to "|" ? For C code styles, we normally did it using "||" to do or check condition.

'||' is conditional. If there are many '||' in 'if', it could be expensive. whilst '|' is not conditional and checks all conditions. So it's for performance purpose.

Reply:
I do not strongly oppose this, but this looks like overdone for performance. for such kind of check, it might be more important for readability. *_*

Reply:
The parameter decoding is on critical path. Thus, I made such optimization.